main/GarnetServer/Program.cs

```diff
@@ -16,82 +12,21 @@ namespace Garnet
     /// </summary>
     class Program
     {
-        private static readonly string CustomRespCommandInfoJsonFileName = "CustomRespCommandsInfo.json";
-
         static void Main(string[] args)
         {
-            try
-            {
-                using var server = new GarnetServer(args);
-
-                // Optional: register custom extensions
-                if (!TryRegisterExtensions(server))
-                {
-                    Console.WriteLine("Unable to register server extensions.");
-                    return;
-                }
-
-                // Start the server
-                server.Start();
+            var builder = Host.CreateEmptyApplicationBuilder(null);
+            builder.Services.AddHostedService(sp => new GarnetService(args));
 
-                Thread.Sleep(Timeout.Infinite);
-            }
-            catch (Exception ex)
+            if (OperatingSystem.IsWindows())
```

Comment:
I think we don't need to guard here as the `UseWindowsService` checks this already.

https://github.com/dotnet/runtime/blob/0b10f4ba0a9174937829ec529c197dd42a569e36/src/libraries/Microsoft.Extensions.Hosting.WindowsServices/src/WindowsServiceLifetimeHostBuilderExtensions.cs#L49-L55
=>
https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.Hosting.WindowsServices/src/WindowsServiceHelpers.cs#L24

Reply:
Resolved.